### PR TITLE
DAOS-9919 test: Don't disable daos repo...

### PR DESCRIPTION
**Pull request by brianjmurrell**
On weekly testing.

Or when an RPM-test-version: pragma has been specified.

### CODE REVIEW

**Review by JohnMalmberg — APPROVED** [review]

**Comment by brianjmurrell** [comment]
FWIW, this PR is required for master branch weekly-testing to be able to provision clusters.

File: ci/provisioning/post_provision_config_common_functions.sh [hunk 1]
```diff
@@ -173,14 +192,14 @@ set_local_repo() {
     version="$(lsb_release -sr)"
     version=${version%%.*}
     if [ "$repo_server" = "artifactory" ] &&
-        [[ $(echo "$COMMIT_MESSAGE" | sed -ne '/^PR-repos: */s/^[^:]*: *//p') = *daos@* ]] ||
-        [[ $(echo "$COMMIT_MESSAGE" |
-             sed -ne "/^PR-repos-$DISTRO: */s/^[^:]*: *//p") = *daos@* ]] ||
-        [ -z "$(echo "$COMMIT_MESSAGE" | sed -ne '/^RPM-test-version: */s/^[^:]*: *//p')" ]; then
-        # Disable the daos repo so that the Jenkins job repo is used for daos packages
+       [ -z "$(rpm_test_version)" ] &&
```
Comment (phender) [inline ci/provisioning/post_provision_config_common_functions.sh:195]: Should this be calling `pr_repos()` as well?

Comment (brianjmurrell) [reply ci/provisioning/post_provision_config_common_functions.sh:195]: No.  Keeping the daos repo enabled even if `PR-repos:` points at a daos PR was a misjudgement.  The only time we want the daos repo enabled is if we want to grab an existing build of daos from it, which only happens when `RPM-test-version:` is used or for the `weekly-testing` runs.

File: ci/provisioning/post_provision_config_common_functions.sh [hunk 2]
```diff
@@ -173,14 +192,14 @@ set_local_repo() {
...
+       [[ $BRANCH_NAME != weekly-testing* ]]; then
```
Comment (brianjmurrell) [inline ci/provisioning/post_provision_config_common_functions.sh:196]: I need to double-check that this is the correct variable.  I might actually want `CHANGE_BRANCH` here, not `BRANCH_NAME`.

Comment (brianjmurrell) [reply ci/provisioning/post_provision_config_common_functions.sh:196]: Given that this only needs to check for the `weekly-testing*` branches and is orthogonal to PR branches, `BRANCH_NAME` should be sufficient here.

**Review by phender — APPROVED** [review]

**Pull request merged**